### DISCUSSION TITLE
Create users and assign rules by sensor nodes

### DISCUSSION THREAD

**Discussion by itdeveloper50** [Security-Onion-Solutions/securityonion#9137] [Unsupported Versions]
We have installed distributed version of Security Onion.
Each branch has its own sensor node.
How to create users and assign roles by their sensor nodes?

For instance, let's say user Alice enters into management (web) and can see only her sensor's alerts.
Same for user Bob, Bob may see only his sensor node's alerts.

Superuser can see all alerts.

**Comment by dougburks** [comment]
Please see the RBAC page in our documentation:
https://docs.securityonion.net/en/2.3/rbac.html

**Reply by itdeveloper50** [reply]
I read it, but there is no exact answer. 
I know the key should be simple.
I am confused now, do not know where to start from.

1. Let's say I have created user.
2. Created role, for example, sensor-analyst.
3. Assigned this role to created user.

After that, how should I manage permissions?

User must be able to see only his own sensor and its alerts.
Your help is appreciated. Thank you!

**Reply by dougburks** [reply]
Have you read through the `Creating Custom Roles` section of that page? It seems to cover your use case. From https://docs.securityonion.net/en/2.3/rbac.html#creating-custom-roles:
> These steps will guide you through an example where we wish to introduce a new role called eastcoast-analyst, which will inherit the same Security Onion permissions as a limited-analyst, but will be restricted to only view a subset of documents in the Elastic stack. We base this role on the limited-analyst instead of the analyst role so that the user does not have the ability to create arbitrary PCAPs on any sensor.

**Reply by itdeveloper50** [reply]
`but will be restricted to only view a subset of documents in the Elastic stack`

Let's say we restricted user to only view a subset of documents in the Elastic stack (only his own sensor),
will the user also be restricted to only view his sensor's alerts in the SO dashboard?
I mean, will it affect also dashboard?

Thank you!

**Reply by TOoSmOotH** [reply]
SOC uses the same permissions as Kibana. So you should only see alerts etc from sensors you are allowed to see.

**Reply by itdeveloper50** [reply]
I created user, created elastic stack role.
But I cannot create SO role.

I cannot find file custom_roles.

Where is it located?
Thank you!

**Reply by itdeveloper50** [reply]
Even if I do like this: `so-user add "testuser@mymail.com" limited-analyst` ,
It creates user named `testuser`, but does not assign role of `limited-analyst`.

**Reply by itdeveloper50** [reply]
When I do `so-eleasticsearch-restart` , at the end it says: `Loading roles...role:created:false, current license is non-compliant for ...`

Is it ok?
Should ignore and not pay attention to that message?

Will I be able to create users according to their sensors?

**Reply by dougburks** [reply]
What version of Security Onion was originally installed and what version are you running now?

**Reply by itdeveloper50** [reply]
@dougburks , we installed `Security Onion-2.3.160-20220829` version originally and still we are running this version.
We use it in airgap mode.

Thank you.

**Reply by dougburks** [reply]
What is the output of the following?
```
sudo grep -A1 auth /opt/so/saltstack/local/pillar/elasticsearch/auth.sls
```

**Reply by itdeveloper50** [reply]
@dougburks , 

```
auth:
    enabled: True
```

**Reply by dougburks** [reply]
OK, that seems to confirm that Elastic auth is enabled. Let's go back to this:

> Even if I do like this: `so-user add "testuser@mymail.com" limited-analyst` ,
> It creates user named `testuser`, but does not assign role of `limited-analyst`.

I just tested on a 2.3.160 installation and this seems to work correctly for me:
```
[doug@securityonion ~]$ sudo so-user list
doug@example.com: superuser 
[doug@securityonion ~]$ sudo so-user add "testuser@mymail.com" limited-analyst
Enter new password:
Syncing users and roles between SOC and Elastic...
Elastic state will be re-applied to affected minions. This may take several minutes...
Successfully added new user to SOC
[doug@securityonion ~]$ sudo so-user list
doug@example.com: superuser 
testuser@mymail.com: limited-analyst 
```

When you do that test, what do you get at the end when running `sudo so-user list`?

You might also want to try a fresh installation (perhaps in a VM) just to see if you get any different results there.

**Reply by itdeveloper50** [reply]
Thank you for reply!

1. I created user testuser@mymail.com and assigned role limited-analyst.
`sudo so-user list` shows it. That is fine.

2. After that, I created limited-analyst.json file in /opt/so/saltstack/local/salt/elasticsearch/roles according to this [link](https://docs.securityonion.net/en/2.3/rbac.html#creating-custom-roles).
I just changed this line: _"query": "{ \"bool\": { \"should\": [ { \"prefix\": { \"observer.name\": \"`myobserver`\" }}.._
4. Ran `sudo salt-call state.highstate`.

Everything is fine, testuser can login into SO and Kibana.
But, when I do `so-elasticsearch-restart`, at the end of message there is one line that bothers me:
`Loading roles..., {"role":{"created:false..... current license is non-compliant for .... field and document level security..role:created:false`

Is it okay? Will testuser be able to see alerts only from myobserver?
Thank you again!
